extension/src/resources/pages/profile.tsx

```diff
@@ -7,20 +7,19 @@ import { Storage } from "@plasmohq/storage";
 import { useStorage } from "@plasmohq/storage/dist/hook";
 
 import type { TwitchUser } from "~types/types";
-import { t } from "~utils/i18nUtils";
 
 type ProfileProps = {
   user: TwitchUser;
 };
 
 export default function Profile({ user }: ProfileProps) {
-  const storage = new Storage();
+  const onStorageClear = new Storage().clear;
   const [currentPronouns] = useStorage("pronouns");
   const [currentOccupation] = useStorage("occupation");
 
   return (
     <div className="max-w-96">
-      <Header />
+      <Header onStorageClear={onStorageClear} />
```

Comment:
Instead of prop drilling the storage clear, what about moving it to the `Header` component? Does it make sense to you?

Because nothing else uses the storage inside the `Profile,` it would make sense if `Profile`  were using it for something else.

Reply:
Prop drilling occur when the prop is passing down throught multiple layers, which is not the case. 

I choose to pass onStorageClear as prop based on:

1.  keep the Header component independet, adaptable to various contexts
2. separation of concerns: Header component does not need to know the details of how storage is cleared, its responsibility is only rendering the UI

@leandrogrillo what do you think? let me know if its make sense